subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -879,51 +939,83 @@ static int broadcast_assistant_reset(struct bap_broadcast_assistant_instance *in
 
 static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
 {
-	(void)broadcast_assistant_reset(&broadcast_assistant);
+	struct bap_broadcast_assistant_instance *inst = inst_by_conn(conn);
+
+	if (!inst) {
+		return;
+	}
+
+	(void)broadcast_assistant_reset(inst);
+}
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	struct bap_broadcast_assistant_instance *inst;
+
+	if (err != 0U) {
+		LOG_DBG("Failed to connect to %p (%u)", conn, err);
+		return;
+	}
+
+	LOG_DBG("Connected: %p", conn);
+
+	inst = &broadcast_assistants[bt_conn_index(conn)];
+
+	if (inst->conn == conn) {
+		return;
+	}
+
+	if (broadcast_assistant_reset(inst) != 0) {
+		LOG_DBG("Failed to reset broadcast assistant: %d", err);
+
+		return;
+	}
+
+	inst->conn = bt_conn_ref(conn);
+	inst->busy = false;
```

Comment:
What is the purpose of this? 

We do not support bonding yet, and even if we did, we cannot rely on the conn index being the same. 

The `if (inst->conn == conn) {` check should always fail as clear the `conn` pointer in the disconnected. 

I don't think we have any use for this function and it should be removed again.

Reply:
here I try to bind a struct bt_conn *conn with a specific bap_broadcast_assistant_instance via the unique conn index.  Since it is not feasible to use CONTAINER_OF() for struct bt_conn *conn to retrieve bap_broadcast_assistant_instance later on.

do you have a better suggesion to bind a struct bt_conn *conn with bap_broadcast_assistant_instance ?

Reply:
> do you have a better suggesion to bind a struct bt_conn *conn with bap_broadcast_assistant_instance ?

The `conn` is bound to the assistant when calling `bt_bap_broadcast_assistant_discover`